### PR TITLE
Video parsing

### PR DESCRIPTION
**Pull request by ArnavAgrawal03**

### CODE REVIEW

File: core/parser/video/parse_video.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+from typing import Dict
+import base64
+from openai import OpenAI
+import assemblyai as aai
+import logging
+from core.models.time_series import TimeSeriesData
+
+logger = logging.getLogger(__name__)
+
+def debug_object(title, obj):
```
Comment (Adityav369) [inline core/parser/video/parse_video.py:11]: LAL

File: core/parser/video/parse_video.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    def get_transcript(self) -> TimeSeriesData:
+        """
+        Get timestamped transcript of the video using AssemblyAI
+
+        Returns:
+            TimeSeriesData object containing transcript
+        """
+        logger.info("Starting video transcription")
+        transcript = self.get_transcript_object()
+        time_to_text = {u.start/1000: u.text for u in transcript.utterances}
```
Comment (Adityav369) [inline core/parser/video/parse_video.py:77]: Pls add comment why 1000 is there

File: core/models/time_series.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        start_idx = max(0, start_idx)
+        end_idx = min(len(self.timestamps) - 1, end_idx)
+        
+        logger.debug(f"Retrieving content between {start_time:.2f}s and {end_time:.2f}s")
+        return [(self.timestamps[i], self.contents[i]) 
+                for i in range(start_idx, end_idx + 1)]
+    
+    def times_for_content(self, content: str) -> List[float]:
+        """Get all timestamps where this content appears"""
+        return self.content_to_times[content]
```
Comment (Adityav369) [inline core/models/time_series.py:77]: format

File: core/parser/combined_parser.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+from .video.parse_video import VideoParser
+
+logger = logging.getLogger(__name__)
+
+class CombinedParser(BaseParser):
+    def __init__(
+        self,
+        unstructured_api_key: str,
+        assemblyai_api_key: str,
+        chunk_size: int = 1000,
```
Comment (Adityav369) [inline core/parser/combined_parser.py:21]: remove defaults

File: core/models/time_series.py (hunk 3 above)
Comment (ArnavAgrawal03) [reply core/models/time_series.py:77]: ran black formatter

File: core/parser/video/parse_video.py (hunk 2 above)
Comment (ArnavAgrawal03) [reply core/parser/video/parse_video.py:77]: added comment

File: core/parser/video/parse_video.py (hunk 1 above)
Comment (ArnavAgrawal03) [reply core/parser/video/parse_video.py:11]: fixed: used black formatter

**Pull request merged**